File: Diagrammatize.Design.Pattern/code/cxx/Observer/A1/IncrementalNumberGenerator.cpp
#include "IncrementalNumberGenerator.hpp"

IncrementalNumberGenerator::IncrementalNumberGenerator(int start, int end_, int inc_):
    number(start), end(end_), inc(inc_) {
}

int IncrementalNumberGenerator::getNumber() {
    return number;
}

void IncrementalNumberGenerator::execute() {
    while (number < end) {
        notifyObservers();
        number += inc;
    }
}

File: Diagrammatize.Design.Pattern/code/python/Observer/A1/IncrementalNumberGenerator.py
from NumberGenerator import NumberGenerator

class IncrementalNumberGenerator(NumberGenerator):
    def __init__(self, start, end, inc):
        super().__init__()
        self.number = start
        self.end = end
        self.inc = inc

    def getNumber(self):
        return self.number

    def execute(self):
        while self.number < self.end:
            self.notifyObservers()
            self.number += self.inc
